ubuntu2404_cuda/Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM ghcr.io/acts-project/ubuntu2404:47
+
+LABEL description="Ubuntu 24.04 with Acts dependencies and CUDA"
+LABEL maintainer="Stephen Nicholas Swatman <stephen.nicholas.swatman@cern.ch>"
+# increase whenever any of the RUN commands change
+LABEL version="1"
+
+# DEBIAN_FRONTEND ensures non-blocking operation (tzdata is a problem)
+ENV DEBIAN_FRONTEND noninteractive
+
+ENV CUDA_RUNFILE_NAME="cuda_12.5.0_555.42.02_linux.run"
+ENV CUDA_INSTALL_PATH="/usr/local/cuda"
+
+RUN apt install wget
+
+RUN wget https://developer.download.nvidia.com/compute/cuda/12.5.0/local_installers/${CUDA_RUNFILE_NAME} && \
+  chmod +x ${CUDA_RUNFILE_NAME} && \
+  ./${CUDA_RUNFILE_NAME} --toolkit --silent --installpath=${CUDA_INSTALL_PATH} && \
+  rm ${CUDA_RUNFILE_NAME}
```

Comment:
I am super not on board with this implementation. Please don't merge stuff right away, and wait for us to have a look first!

You'll need to switch to the same setup that I use for all the other GPU supporting images. Installing CUDA/HIP/oneAPI using APT. Since that's the one technique that all of them support. So let's not install each of them in a different way!

I.e. go and modify this image like: https://github.com/acts-project/machines/blob/master/ubuntu2004_cuda/Dockerfile